torch/nn/modules/module.py

```diff
@@ -1059,9 +1059,9 @@ def _call_impl(self, *input, **kwargs):
         if self._backward_hooks or _global_backward_hooks:
             full_backward_hooks, non_full_backward_hooks = self._get_backward_hooks()
         if _global_forward_pre_hooks or self._forward_pre_hooks:
-            for hook in itertools.chain(
+            for hook in list(itertools.chain(
```

Comment:
Instead of using `list()` + `itertools.chain()`, can you please try:
```python
for hook in (*_global_forward_pre_hooks.values(), *self._forward_pre_hooks.values()):
```